src/v/security/sasl_authentication.h

```diff
@@ -58,11 +87,15 @@ class sasl_server final {
     }
 
     void set_mechanism(std::unique_ptr<sasl_mechanism> m) {
-        vassert(!_mechanism, "Cannot change mechanism");
+        // TODO(oren): tidy
+        // vassert(!_mechanism, "Cannot change mechanism");
```

Comment:
Should we blow away the sasl_server instance instead? that seems a bit heavier. maybe a check that they are the same type of mechanism if available?